### ISSUE TITLE
docker retracer shouldn't crash tracing if there is no docker

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/niceman#212]
```shell
(niceman-dev) [yhalchen@discovery niceman]$ niceman retrace /usr/bin/python2
2018-04-13 13:18:56,494 [INFO   ] Entering iteration #1 over Tracers 
2018-04-13 13:18:56,501 [INFO   ] DebTracer: 0 envs with 1 other files remaining 
2018-04-13 13:18:56,527 [INFO   ] CondaTracer: 0 envs with 1 other files remaining 
2018-04-13 13:18:56,548 [INFO   ] VenvTracer: 0 envs with 1 other files remaining 
2018-04-13 13:18:56,573 [INFO   ] VCSTracer: 0 envs with 1 other files remaining 
2018-04-13 13:18:56,586 [ERROR  ] Failed to start ['docker', 'image', 'inspect', u'/usr/bin/python2']'': [Errno 2] No such file or directory [subprocess.py:_execute_child:1025] 
2018-04-13 13:18:56,588 [ERROR  ] [Errno 2] No such file or directory [subprocess.py:_execute_child:1025] (OSError) 
```
I guess there should first be a test if tracer is "available" for an environment with a single warning about which aren't, and then they would be excluded from a list of tracers.  Assigning to both @kyleam and @mjtravers for a healthy collaboration/competition ;)

**Comment by yarikoptic** [comment]
and then there should be a general handling of unexpected behavior of a tracer - e.g. if it thought it would work but crashes - issue a warning and proceed to the next one, without causing the entire process to collapse.  I've made a dummy `docker` script which just echos empty `{}`, which is legit JSON, so it now crashes with
```shell
(niceman-dev) [yhalchen@discovery niceman]$ niceman --dbg retrace /usr/bin/python2
2018-04-13 13:22:41,978 [INFO   ] Entering iteration #1 over Tracers 
2018-04-13 13:22:41,986 [INFO   ] DebTracer: 0 envs with 1 other files remaining 
2018-04-13 13:22:42,010 [INFO   ] CondaTracer: 0 envs with 1 other files remaining 
2018-04-13 13:22:42,032 [INFO   ] VenvTracer: 0 envs with 1 other files remaining 
2018-04-13 13:22:42,056 [INFO   ] VCSTracer: 0 envs with 1 other files remaining 
Traceback (most recent call last):
  File "/ihome/yhalchen/.conda/envs/niceman-dev/bin/niceman", line 11, in <module>
    load_entry_point('niceman', 'console_scripts', 'niceman')()
  File "/ihome/yhalchen/proj/repronim/niceman/niceman/cmdline/main.py", line 238, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/ihome/yhalchen/proj/repronim/niceman/niceman/interface/base.py", line 352, in call_from_parser
    return cls.__call__(**kwargs)
  File "/ihome/yhalchen/proj/repronim/niceman/niceman/interface/retrace.py", line 98, in __call__
    session=session
  File "/ihome/yhalchen/proj/repronim/niceman/niceman/interface/retrace.py", line 186, in identify_distributions
    files_to_trace):
  File "/ihome/yhalchen/proj/repronim/niceman/niceman/distributions/docker.py", line 124, in identify_distributions
    'image', 'inspect', file])[0])[0]
KeyError: 0

```

**Comment by kyleam** [comment]
Should be fixed with #219.
